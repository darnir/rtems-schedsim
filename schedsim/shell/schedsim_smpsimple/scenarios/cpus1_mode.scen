#
#  Check increasing and decrease priority self with no change
#
rtems_init 1
task_create -p smp1 5
# GOAL: 0x0a010001 is heir and executing on CPU 0
cpus
task_create smp2 4
# GOAL: 0x0a010001 is executing and 0x0a010002 is heir on CPU 0
cpus

task_mode -P
# GOAL: 0x0a010002 is heir and executing on CPU 0
cpus
