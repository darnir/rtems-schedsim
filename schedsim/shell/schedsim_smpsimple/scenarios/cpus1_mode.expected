Script File               : scenarios/cpus1_mode.scen
verbose                   : 1
==> 1: #
==> 2: #  Check increasing and decrease priority self with no change
==> 3: #
==> 4: rtems_init 1
  Thread Heir: 0x09010001 priority=255
  Thread Executing: 0x09010001 priority=255
==> 5: task_create -p smp1 5
Creating task with:  Preemption: no Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=5
Task (smp1) starting: id=0x0a010001, priority=5
  Thread Heir: 0x0a010001 priority=5
  Thread Executing: 0x0a010001 priority=5
==> 6: # GOAL: 0x0a010001 is heir and executing on CPU 0
==> 7: cpus
=== CPU Status
            EXECUTING    /     HEIR         /    IDLE        /  SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5 / 0x09010001 @255         false
=== End of Ready Set of Threads
==> 8: task_create smp2 4
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=4
Task (smp2) starting: id=0x0a010002, priority=4
  Thread Heir: 0x0a010002 priority=4
==> 9: # GOAL: 0x0a010001 is executing and 0x0a010002 is heir on CPU 0
==> 10: cpus
=== CPU Status
            EXECUTING    /     HEIR         /    IDLE        /  SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010002 @  4 / 0x09010001 @255         false
=== End of Ready Set of Threads
==> 11: 
==> 12: task_mode -P
  Thread Executing: 0x0a010002 priority=4
Previous Mode: Preemption: no Timeslicing: no
Current Mode:  Preemption: yes Timeslicing: no
==> 13: # GOAL: 0x0a010002 is heir and executing on CPU 0
==> 14: cpus
=== CPU Status
            EXECUTING    /     HEIR         /    IDLE        /  SWITCH NEEDED
  CPU 0: 0x0a010002 @  4 / 0x0a010002 @  4 / 0x09010001 @255         false
=== End of Ready Set of Threads
